aten/src/ATen/TensorOptions.h

```diff
@@ -205,4 +220,28 @@ inline TensorOptions device_index(int32_t device_index) {
 inline TensorOptions requires_grad(bool requires_grad = true) {
   return TensorOptions().requires_grad(requires_grad);
 }
+
+// From Tensor.h:
+
+inline Tensor Tensor::to(TensorOptions options, bool non_blocking) {
+  DeviceGuard guard(options.device());
+  auto copy = options.type().copy(*this, non_blocking);
+  if (copy.is_variable()) {
+    copy.set_requires_grad(options.requires_grad());
```

Comment:
`to` doesn't have a `requires_grad` parameter in PyTorch, and I don't think it should; it's not a tensor creation function any more than `add` is.  Why do we have inconsistency is this direction over say, none of the tensor creation functions take requires_grad, and you have to call `requires_grad_()` at the end of a call chain (why is the recommended way to do this in PyTorch now).

Reply:
Thanks for pointing this out, I will make it consistent with PyTorch!

Reply:
Although I guess I was concerned that people would write `to(tensor.options().requires_grad())` and then be confused that the `requires_grad` has no effect. This is not an issue in PyTorch since `to()` does not have `requires_grad`, but it's a bit hard to work around here. Do you think we should add a check for `requires_grad` being set in the `TensorOptions` and warning the user that is ignored? I feel like it would be better to avoid this unexpected behavior and just set `requires_grad` to that of the `TensorOptions`.